casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -320,21 +320,27 @@ object ProtoUtil {
     } yield ancestorChannels
 
   def chooseNonConflicting[F[_]: Monad: Log: BlockStore](
-      blocks: Seq[BlockMessage],
+      blockHashes: Seq[BlockHash],
       dag: BlockDagRepresentation[F]
   ): F[Seq[BlockMessage]] = {
     def nonConflicting(b: BlockMessage): BlockMessage => F[Boolean] =
       conflicts[F](_, b, dag).map(b => !b)
 
-    blocks.toList
-      .foldM(List.empty[BlockMessage]) {
-        case (acc, b) =>
-          Monad[F].ifM(acc.forallM(nonConflicting(b)))(
-            (b :: acc).pure[F],
-            acc.pure[F]
-          )
-      }
-      .map(_.reverse)
+    for {
+      blocks <- blockHashes.toList.traverse(hash => ProtoUtil.unsafeGetBlock[F](hash))
+      result <- blocks
+                 .foldM(List.empty[BlockMessage]) {
+                   case (acc, b) =>
+                     acc.forallM(nonConflicting(b)).map { isNonConflictingWithAll =>
```

Comment:
the previous version seemed easier to read. why the change?

Reply:
I thought this version was easier to read; the Monad[F].ifM syntax just doesn't read cleanly to me.

Reply:
I agree with Dom that the previuos was cleaner. And even if ifM is difficult to read, because of the fact that it's a standard operator it is easy to learn once and then expect the same behaviour everywhere

Reply:
Ok I'll revert to using ifM.